File: Bmi_Laravel/app/Models/BmiRecord.php
<?php
namespace App\Models;
use Illuminate\Database\Eloquent\Model;

class BmiRecord extends Model
{
      protected $table = 'bmi_records';
    protected $fillable = [
        'user_id',
        'height',
        'weight',
        'bmi',
        'category',
        'recorded_at',
        'age',
        'gender',
        'bmr',
        'activity_level', 
        'tdee' 
    ];
    public function user()
    {
        return $this->belongsTo(User::class);
    }

    

    public function recommendations()
    {
        return $this->hasMany(Recommendation::class);
    }
}

File: bmi_app/lib/models/bmi_record.dart
class BmiRecord {
  final int id;
  final int userId;
  final double height;
  final double weight;
  final double bmi;

  BmiRecord({
    required this.id,
    required this.userId,
    required this.height,
    required this.weight,
    required this.bmi,
  });

  factory BmiRecord.fromJson(Map<String, dynamic> json) {
    return BmiRecord(
      id: json['id'],
      userId: json['user_id'],
      height: json['height'].toDouble(),
      weight: json['weight'].toDouble(),
      bmi: json['bmi'].toDouble(),
    );
  }
}